addons/mail/controllers/thread.py

```diff
@@ -70,6 +71,25 @@ def read_subscription_data(self, follower_id):
     def _get_allowed_message_post_params(self):
         return {"attachment_ids", "body", "message_type", "partner_ids", "subtype_xmlid", "parent_id"}
 
+    def modify_anchor_elements(self, body):
+        root = html.fromstring(body)
+        for a in root.xpath("//a[contains(@class, 'o_message_redirect')]"):
+            message_id = request.env["mail.message"].sudo().browse(int(a.get("data-oe-id")))
+            if message_id.exists():
+                thread_id = request.env[message_id.model].sudo().browse(message_id.res_id)
+                a.text = None
+                for child in list(a):
+                    a.remove(child)
+                display_name = thread_id.display_name or f"{request.env['ir.model']._get(thread_id._name).display_name} ({thread_id.id})"
+                a.extend([
+                    html.builder.SPAN(f"#{display_name}"),
+                    html.Element("i", {"class": "fa fa-angle-right ps-2 pe-1"}),
+                    html.Element("i", {"class": "fa fa-comment"})
+                ])
```

Comment:
Do not stylize in python + stored in DB. Prefer to keep the content as simple as you can.
For example have it `.o_message_redirect[data-oe-id]`.
If you need to also store thread model & id while passing the html sanitizer of body, you could solve this by nesting and combining with many `data-oe-id`/`data-oe-model`, both of which are ok for html sanitizer. For example:

```xml
<a class=".o_message_redirect" data-oe-id="messageId"><a data-oe-model="threadModel" data-oe-id="threadId"></a>
```

Sure, not the prettiest, but at least the storing is minimal and we are not stuck with a styling that we cannot properly adapt / change / migrate.

The styling of this part of message should be made completely in frontend in JS.
(note that sending it by email can stay with the not-so-nice UI, but if you really want a nice UI for that case too, prefer to render it just before sending it by email, and the email content is rendered with nice UI but the message body in DB stays without the style)

Reply:
I did think of storing the `id` and `model` in element `dataSet` but, the issue with this is that I will have to fetch the thread data from server as it would be the case a lot time when the referenced thread data won't be present.
So, It was a tradeoff by losing the flexibility to update the reference name and styling in client we would save us from fetching the thread every time and delaying the display of the message.

Reply:
@rapr-odoo A solution to this problem is to pass the thread name in `message_format`